Zotero/Scenes/Detail/SingleCitation/Views/SingleCitationViewController.swift

```diff
@@ -164,64 +236,66 @@ final class SingleCitationViewController: UIViewController {
                 }
             }
         }
-
-        if state.changes.contains(.preview) {
-            updatePreferredContentSize()
-        }
-
-        if state.changes.contains(.copied) {
-            navigationController?.presentingViewController?.dismiss(animated: true, completion: nil)
-        }
-
-        func updatePreview(isLoading: Bool) {
-            guard previewContainer.isHidden != isLoading else { return }
-
-            activityIndicatorContainer.isHidden = !isLoading
-            previewContainer.isHidden = isLoading
-            locatorButton.isEnabled = !isLoading
-            locatorTextField.isEnabled = !isLoading
-        }
     }
 
+    // MARK: - Helpers
     private func setupRightButtonItem(isLoading: Bool) {
         guard navigationItem.rightBarButtonItem == nil || isLoading == (navigationItem.rightBarButtonItem?.customView == nil) else { return }
-
         if isLoading {
             let indicator = UIActivityIndicatorView(style: .medium)
             indicator.startAnimating()
             navigationItem.rightBarButtonItem = UIBarButtonItem(customView: indicator)
         } else {
             let copy = UIBarButtonItem(title: L10n.copy, style: .done, target: nil, action: nil)
             copy.rx.tap.subscribe(onNext: { [weak self] in
-                self?.viewModel.process(action: .copy)
+                guard let self else { return }
+                viewModel.process(action: .copy(webView: previewWebView))
             })
             .disposed(by: disposeBag)
             navigationItem.rightBarButtonItem = copy
         }
     }
 
-    // MARK: - Helpers
-    private func localized(locator: String) -> String {
-        return NSLocalizedString("citation.locator.\(locator)", comment: "")
-    }
-
     private func updatePreferredContentSize() {
-        let size = view.systemLayoutSizeFitting(CGSize(width: SingleCitationViewController.width, height: .greatestFiniteMagnitude))
-        preferredContentSize = CGSize(width: SingleCitationViewController.width, height: size.height - view.safeAreaInsets.top)
+        let width = SingleCitationViewController.width
+        let height = collectionView.collectionViewLayout.collectionViewContentSize.height
+        preferredContentSize = CGSize(width: width, height: height)
         navigationController?.preferredContentSize = preferredContentSize
     }
 }
 
 extension SingleCitationViewController: WKScriptMessageHandler {
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
         guard message.name == "heightHandler", let height = message.body as? CGFloat else { return }
-        webViewHeight.constant = height
+        viewModel.process(action: .setPreviewHeight(height))
```

Comment:
In my tests, the `height` value is always the same `20`. Can you check if there is an issue here?

Reply:
You're right, height was not calculated properly. The web view which did the citation didn't have the same width as web view that's used to show the citation, so the height differed. I also synced their styles so that the same fonts are used in both. Fixed in 1d2d7ab521af4fb12717934f4d8cdfb22b7e657b.